### PR TITLE
Fix parsing of empty RFC5424 structured data in a message.

### PR DESCRIPTION
**Pull request by bernd**
This avoids throwing an exception and also strips the `- - ` from the resulting message.

Refs Graylog2/graylog2-server#1161

### CODE REVIEW

**Comment by kroepke** [comment]
lgtm

File: src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java [hunk 1]
```diff
@@ -84,6 +87,14 @@ public static StructuredSyslogMessage fromString(
 
     private void  deserialize(final String stringMessage) {
 
+        // Check if the RFC5424 MSGID and STRUCTURED-DATA fields are empty.
+        // This avoids throwing an exception and also strips the "- - " from the message.
+        // See: https://github.com/Graylog2/graylog2-server/issues/1161
+        if (stringMessage.startsWith(EMPTY_STRUCTURED_DATA_PREFIX)) {
```
Comment (joschi) [inline src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java]: A little late to the party, sorry.

Shouldn't we check if `message` is null or empty and that `structuredData` is an empty collection? Otherwise messages which start with "- - " intentionally will incorrectly get truncated.

Comment (bernd) [inline src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java]: No need to excuse, we merged that pretty quickly. :wink: Thank you for still looking at it!

At this point we know that this is a RFC5424 message and so there has to be a `MSGID` and `STRUCTURED-DATA` section. Since both can be `NIL` (`-`), we quickly check for that.

At least that is my understanding. Did I miss something?

Comment (joschi) [inline src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java]: You're correct. I was wrongly assuming that RFC 3164 messages could also be an input for that method.

I've added a test in commit 6d1294cf7ec862aa62ba3a186b10e2663d447311 to test my imagined corner-case specifically.

Comment (bernd) [inline src/main/java/org/graylog2/syslog4j/impl/message/structured/StructuredSyslogMessage.java]: Awesome, thank you! :+1:
